frontend/src/components/SyncButton.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <div
+    v-if="!isLogin && user"
+    class="col-2 button-container"
+  >
+    <div v-if="inDrivesView">
+      <div v-if="syncStatus === 'untouched'">
+        <svg
+          class="icon"
+          @click="triggerSync"
+        >
+          <use xlink:href="../assets/sprite.svg#sync-alt" />
+        </svg>
+      </div>
+      <div v-else-if="syncStatus === 'loading'">
+        <div class="spinner-border icon" />
+      </div>
+      <div v-else-if="syncStatus === 'success'">
+        <svg class="icon success">
+          <use xlink:href="../assets/sprite.svg#success" />
+        </svg>
+      </div>
+      <div v-else-if="syncStatus === 'fail'">
+        <svg class="icon fail">
+          <use
+            xlink:href="../assets/sprite.svg#fail"
+            href="../assets/sprite.svg#fail"
+          />
+        </svg>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { IS_ONLINE, SYNC } from '@/store/constants';
+import { USER } from '@/store';
+import { mapActions, mapState } from 'vuex';
+import { driveListRoute, loginRoute } from '@/router/routes';
+
+export default {
+  name: 'SyncButton',
+  computed: {
+    ...mapState([IS_ONLINE, USER]),
+  },
+  data() {
+    return {
+      isLogin: this.$router.currentRoute.path === loginRoute.path,
+      inDrivesView: false,
+      syncStatus: 'untouched',
+      watchingIsOn: false,
+    };
+  },
+  methods: {
+    ...mapActions({ sync: SYNC }),
+    async triggerSync() {
+      this.sync();
+      this.syncStatus = 'loading';
+      this.$store.subscribe(mutation => {
+        if (mutation.type === 'drives/setData') {
+          this.syncStatus = 'success';
+          this.unsubscribe();
+        }
+      });
+    },
```

Comment:
I have some troubles here. 

I would like to watch for mutation rather than api response. The logic responsible for commiting that mutation covers a lot of different cases and it seems unnecessary to reproduce it. So what I would want to do is:

1. trigger sync and cause "loading"
1. resolve function with "success" when mutation is committed
1. if it's not happening for 30 seconds return "fail"
1. Wait 3sec to revert button to neutral state

I tried to use setTimeout and Promise but apparently I don't know these functions well enough to make it work. If anybody has an idea how to sensibly glue it together I'll be much obliged.

Reply:
Ok I managed to write it down. Don't bother helping with this unless you see a better solution :)